spec/models/course/lesson_plan/lesson_plan_item_spec.rb

```diff
@@ -99,5 +99,63 @@
         end
       end
     end
+
+    describe 'personal times' do
+      let(:student1) { create(:course_student, course: course) }
+      let(:student2) { create(:course_student, course: course) }
+      let(:personal_time) do
+        personal_time = lesson_plan_item.find_or_create_personal_time_for(student1)
+        personal_time.save!
+        personal_time
+      end
+
+      it 'creates a personal time for course_user' do
+        student1
+        student2
+        expect(personal_time.course_user).to eq student1
+        expect(personal_time.lesson_plan_item).to eq lesson_plan_item
+      end
+
+      it 'eager loads personal times for course_user' do
+        personal_time
+        items = Course::LessonPlan::Item.where(id: lesson_plan_item).with_personal_times_for(student1).to_a
+        expect(items.first.personal_times.loaded?).to be true
+        expect(items.first.personal_time_for(student1)).to eq personal_time
+      end
+
+      it 'does not eager load personal times for another course_user' do
```

Comment:
For this scenario, I think it's better that student2 also has a personal time on the same item. Then the test should be that student2's personal time gets loaded but not student1's.